<launch>
  <!--
      online segmentation using .npy & SqueezeSeg model
      this script can
                    1. read all .npy file from lidar_2d folder
                    2. predict label from SqueezeSeg model using tensorflow
                    3. publish to 'sqeeuze_seg/points' topic
  -->
  <param name="npy_path" type="string" value="/media/cyber-z/E/npy/" />
  <param name="npy_file_list" type="string" value="$(find squeezeseg_ros)/script/data/ImageSet/test.txt" />
  <param name="read_list" type="bool" value="True" />
  <param name="read_number" type="int" value="77" />
  <param name="pub_topic" type="string" value="/squeeze_seg/points" />
  <param name="fov" type="string" value="-45,45" />
  <!-- should be the path for checkpoint -->
  <param name="input_type" type="string" value="npy" />
  <param name="checkpoint" type="string" value="$(find squeezeseg_ros)/script/data/SqueezeSeg/model.ckpt-23000" />
  <param name="gpu" type="string" value="-1" />

  <node pkg="squeezeseg_ros" type="squeezeseg_ros_node.py" name="squeezeseg_ros_node" output="screen" />

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find squeezeseg_ros)/rviz/squeezeseg_ros.rviz" />

</launch>


